### PR TITLE
Bristol brca refactoring

### PR DESCRIPTION
**Pull request by FrancescoNea**
Refactored Bristol (RVJ) BRCA handler and coverage test

### CODE REVIEW

File: test/lib/import/brca/providers/bristol/bristol_handler_test.rb [hunk 1]
```diff
@@ -10,21 +10,47 @@ def setup
...
     assert_equal 'c.8167G>C', @genotype.attribute_map['codingdnasequencechange']
   end
 
+  test 'process_mutated_and_normal_gene' do
+    @handler.process_cdna_change(@genotype, @record)
+    res = @handler.process_gene(@genotype, @record)
+    assert_equal 2, res.size
+    assert_equal 1, res[0].attribute_map['teststatus']
+    assert_equal 2, res[1].attribute_map['teststatus']
+    # assert_equal 'c.8167G>C', @genotype.attribute_map['codingdnasequencechange']
```
Comment (ollietulloch) [inline test/lib/import/brca/providers/bristol/bristol_handler_test.rb:34]: @FrancescoNea please remove commented out code if its no longer needed

File: lib/import/helpers/brca/providers/rvj/rvj_constants.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+module Import
+  module Helpers
+    module Brca
+      module Providers
+        module Rvj
+          # Constants used by LondonKgcHandlerColorectal
```
Comment (ollietulloch) [inline lib/import/helpers/brca/providers/rvj/rvj_constants.rb:6]: @FrancescoNea is this comment correct? The branch name references Bristol

**Review by ollietulloch — CHANGES_REQUESTED** [review]
Hi @FrancescoNea,

A couple of minor things to change

File: lib/import/helpers/brca/providers/rvj/rvj_constants.rb (hunk 2 above)
Comment (FrancescoNea) [reply lib/import/helpers/brca/providers/rvj/rvj_constants.rb:6]: Hi @ollietulloch sorry, that comment was somehow inherited from another extractor. Now corrected.

File: test/lib/import/brca/providers/bristol/bristol_handler_test.rb (hunk 1 above)
Comment (FrancescoNea) [reply test/lib/import/brca/providers/bristol/bristol_handler_test.rb:34]: Hi @ollietulloch, all unused code now removed

File: lib/import/brca/providers/bristol/bristol_handler.rb [hunk 3]
```diff
@@ -6,43 +6,67 @@ module Providers
...
+            return if record.raw_fields['gene'].nil?
+
+            genotypes = []
+            negative_gene = %w[BRCA1 BRCA2] - [record.raw_fields['gene']]
+            process_negative_record(record, negative_gene, genotypes)
+            genotype.add_gene(record.raw_fields['gene'])
+            genotypes.append(genotype)
+          end
+
+          # rubocop:disable Style/GuardClause
```
Comment (ollietulloch) [inline lib/import/brca/providers/bristol/bristol_handler.rb:60]: @FrancescoNea I'm not sure that this cop needs to be disabled, it looks correctable?

File: lib/import/brca/providers/bristol/bristol_handler.rb [hunk 4]
```diff
@@ -6,43 +6,67 @@ module Providers
...
+                                                       record.raw_fields,
+                                                       PASS_THROUGH_FIELDS)
+            genotypes.append(duplicated_genotype)
+          end
+
+          def process_gene(genotype, record)
+            return if record.raw_fields['gene'].nil?
+
+            genotypes = []
+            negative_gene = %w[BRCA1 BRCA2] - [record.raw_fields['gene']]
```
Comment (ollietulloch) [inline lib/import/brca/providers/bristol/bristol_handler.rb:54]: @FrancescoNea `record.raw_fields['gene']` is repeated in this method, you could assign it to a `raw_gene` variable, for example.

**Review by ollietulloch — CHANGES_REQUESTED** [review]
Hi @FrancescoNea,

A couple additional minor changes please

Thanks

File: lib/import/brca/providers/bristol/bristol_handler.rb (hunk 4 above)
Comment (FrancescoNea) [reply lib/import/brca/providers/bristol/bristol_handler.rb]: record.raw_fields['gene'] now assigned to positive_raw_gene

**Pull request merged**